src/StackAdmin/Azs.Storage.Admin/Module/Azs.Storage.Admin/Generated.PowerShell.Commands/SwaggerPathCommands/Remove-AzsStorageQuota.ps1

```diff
@@ -69,57 +78,65 @@ function Remove-AzsStorageQuota {
     }
 
     Process {
+    
+        $ErrorActionPreference = 'Stop'
+
+        $quotaName = $Name
 
-        if ('ResourceId' -eq $PsCmdlet.ParameterSetName) {
+        if ('InputObject' -eq $PsCmdlet.ParameterSetName -or 'ResourceId' -eq $PsCmdlet.ParameterSetName) {
             $GetArmResourceIdParameterValue_params = @{
                 IdTemplate = '/subscriptions/{subscriptionId}/providers/Microsoft.Storage.Admin/locations/{location}/quotas/{quotaName}'
             }
-            $GetArmResourceIdParameterValue_params['Id'] = $ResourceId
+
+            if ('ResourceId' -eq $PsCmdlet.ParameterSetName) {
+                $GetArmResourceIdParameterValue_params['Id'] = $ResourceId
+            }
+            else {
+                $GetArmResourceIdParameterValue_params['Id'] = $InputObject.Id
+            }
             $ArmResourceIdParameterValues = Get-ArmResourceIdParameterValue @GetArmResourceIdParameterValue_params
             $location = $ArmResourceIdParameterValues['location']
 
-            $Name = $ArmResourceIdParameterValues['quotaName']
-        } else {
-            $Name = Get-ResourceNameSuffix -ResourceName $Name
```

Comment:
WHy are you removing the call to Get-ResourceNameSuffix

Reply:
The quotaName is already parsed by "Get-ArmResourceIdParameterValue", seems it's not necessary to call "Get-ResourceNameSuffix" this time.